nemo/export/tensorrt_llm.py

```diff
@@ -0,0 +1,707 @@
+        """
+
+        self.model_dir = model_dir
+        self.lora_ckpt_list = lora_ckpt_list
+        self.model = None
+        self.tokenizer = None
+        self.n_gpus = None
+        self.config = None
+        self.ptuning_tables = []
+        self.p_table = None
+        self.task_vocab_size = 0
+        self.task_ids = {}
+
+        if load_model:
+            self._load()
+
+    def export(
+        self,
+        nemo_checkpoint_path: str,
+        model_type: str,
+        delete_existing_files: bool = True,
+        n_gpus: int = 1,
+        tensor_parallel_size: int = None,
+        pipeline_parallel_size: int = None,
+        max_input_token: int = 256,
+        max_output_token: int = 256,
+        max_batch_size: int = 8,
+        max_prompt_embedding_table_size=None,
+        use_inflight_batching: bool = False,
+        enable_context_fmha: bool = True,
+        paged_kv_cache: bool = False,
+        dtype: str = "bfloat16",
+        load_model: bool = True,
+        enable_multi_block_mode: bool = False,
+        use_lora_plugin: str = None,
+        lora_target_modules: List[str] = None,
+        max_lora_rank: int = 64,
+        save_nemo_model_config: bool = False,
+    ):
+        """
+        Exports nemo checkpoints to TensorRT-LLM.
+
+        Args:
+            nemo_checkpoint_path (str): path for the nemo checkpoint.
+            model_type (str): type of the model. Currently, "llama", "gptnext", "falcon", and "starcoder" are supported.
+            delete_existing_files (bool): if Truen, deletes all the files in model_dir.
+            n_gpus (int): number of GPUs to use for inference.
+            tensor_parallel_size (int): tensor parallelism.
+            pipeline_parallel_size (int): pipeline parallelism.
+            max_input_token (int): max input length.
+            max_output_token (int): max output length.
+            max_batch_size (int): max batch size.
+            max_prompt_embedding_table_size (int): max prompt embedding size.
+            use_inflight_batching (bool): if True, enables inflight batching for TensorRT-LLM Triton backend.
+            enable_context_fmha (bool): if True, use fused Context MultiHeadedAttention.
+            paged_kv_cache (bool): if True, uses kv cache feature of the TensorRT-LLM.
+            dtype (str): Floating point type for model weights (Supports BFloat16/Float16).
+            load_model (bool): load TensorRT-LLM model after the export.
+            enable_multi_block_mode (bool): enable faster decoding in multihead attention. Required for long context.
+        """
+
+        if model_type not in self.get_supported_models_list:
+            raise Exception(
+                "Model {0} is not currently a supported model type. "
+                "Supported model types are llama, gptnext, falcon, and starcoder".format(model_type)
+            )
+
+        if model_type == "gpt" or model_type == "starcoder":
```

Comment:
Move arg cleanup to another function

Reply:
Only doing it in that location. Makes sense to move it to another function if we need to reuse it, I think.